sympy/vector/vector.py

```diff
@@ -344,6 +344,70 @@ def _div_helper(one, other):
         else:
             raise TypeError("Invalid division involving a vector")
 
+    def equals(self, other):
+        """
+        Check if the scalar components of self & others's vector system
+        are equal after simplification.
+        Parameters:
+        - self: The first vector for comparison.
+        - others: The second vector for comparison.
+        - N: instance of a 3D coordinate system.
+        Returns:
+        - True if the scalar components are equal after simplification,
+        otherwise False. If an error occurs during the process, a
+        RuntimeError is raised with an error message.
+        """
+        import sympy
+        from sympy import simplify, radsimp
+        from sympy.vector import Vector
+
+        # finding vector components of self and other with their respective coordinate system
+        V_S = [0, 0, 0]
+        V_O = [0, 0, 0]
+
+        # finding scalar components of self and other
+        S_S = [0, 0, 0]
+        S_O = [0, 0, 0]
+        if self == Vector.zero:
+            V_S = [0, 0, 0]
+            S_S = [0, 0, 0]
+        else:
+            index = 0
+            for i in self.components:
+                if isinstance(i, sympy.vector.vector.BaseVector):
+                    V_S[index] = i
+                    S_S[index] = self.components[i]
+                    index += 1
+
+        if other == Vector.zero:
+            V_O = [0, 0, 0]
+            S_O = [0, 0, 0]
+        else:
+            index = 0
+            for i in other.components:
+                if isinstance(i, sympy.vector.vector.BaseVector):
+                    V_O[index] = i
+                    S_O[index] = other.components[i]
+                index += 1
```

Comment:
I don't know if this code is correct but I am sure that there is a better way to express both vectors in the same coordinate system. There should be an existing function for doing this.

Reply:
there is a function used like, `CoordSys3D('self')`, but it has some drawbacks for the purpose in PR/issue/tests written, like:
It is only to check whether vectors are in same coordinate system or not. With some limits:
  * Q = CoordSys3D('Q')
    i, j, k = Q.base_vectors()
    A = sympy.sqrt(2) + sympy.sqrt(6)) / (sympy.sqrt(sympy.sqrt(3) + 2))
    v1 = 2*i +  2*j
    v2 =  A*i + 2*j
    #when we do use `CoordSys3D`, it will transform like:
    v1 = 2*Q.i + 2*Q.j
    v2 = ((sqrt(2) + sqrt(6))/sqrt(sqrt(3) + 2))*Q.i + 2*Q.j

    Can we compare them without using components for coordinate systems? No. 

    Why? As the expressions are not in simplified form, and we cannot perform `radsimp` on `CoordSys3D`

    Can we use equals method on them? No. CoordSys3D do not support equals.

    Why not using `components` on CoordSys3D and compare keys and values using equals? Again, CoordSys3D do 
    not support `components`.

    Why not using `args`? Example if v1 = Q.i, then v1.args returns a tuple (0, Q), and when for loop used on it scaler 
    components comes to be `0` and vector comes to be coordinate system not proper vector (which happens when 
    there is a scaler component of a vector except 1). And for Vector.zero, it returns an empty tuple, again not so much 
    helpful.

So, the best way is using loops on `components`, to find the scaler and vector components. And if a vector is zero (like Vector.zero), simply returning a list of [0, 0, 0]. 


With your experience, if you have a better method to do it, please let me know. Thanks.

Reply:
I don't have much experience with the vector module so I don't know what the right way to do this is but it is definitely not the code that is here. There are existing functions for expressing a vector in different coordinate systems and those should be used.